lib/client/tcp-client.js

```diff
@@ -30,27 +30,33 @@ TcpClient.prototype.connect = function(callback) {
       self.responseBuffer += data.toString();
       if (self.responseBuffer.substring(self.responseBuffer.length - 2, self.responseBuffer.length) == FS + CR) {
         var ack = self.parser.parse(self.responseBuffer.substring(1, self.responseBuffer.length - 2));
-        self.callback(null, ack);
         self.responseBuffer = "";
         self.awaitingResponse = false;
         if (!self.keepalive) {
           self.close();
         }
+        self.callback(null, ack);
       }
     });
     callback();
   });
+  self.client.on('close', function() {
+    self.responseBuffer = "";
+    self.awaitingResponse = false;
+    delete self.client;
+  })
   self.client.on('error', function(err) {
     callback(err);
   })
 }
 
 TcpClient.prototype.send = function(msg, callback) {
   var self = this;
-  self.callback = callback || self.callback;
   if (self.awaitingResponse) {
-    self.callback(new Error("Can't send while awaiting response"));
+    callback(new Error("Can't send while awaiting response"));
```

Comment:
replaced the callback, so that the first message callback didn't receive the ack, but the second one did